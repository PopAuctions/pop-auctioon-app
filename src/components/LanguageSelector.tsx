import React from 'react';
import { View, Text, TouchableOpacity, StyleSheet } from 'react-native';
import { useTranslation } from '../hooks/useTranslation';

export default function LanguageSelector() {
  const { t, locale, changeLanguage } = useTranslation();

  const languages = [
    { code: 'es', name: 'EspaÃ±ol', flag: 'ðŸ‡ªðŸ‡¸' },
    { code: 'en', name: 'English', flag: 'ðŸ‡ºðŸ‡¸' },
  ];

  return (
    <View style={styles.container}>
      <Text style={styles.title}>{t('account.language')}</Text>
      <View style={styles.languageList}>
        {languages.map((language) => (
          <TouchableOpacity
            key={language.code}
            style={[
              styles.languageItem,
              locale === language.code && styles.selectedLanguage,
            ]}
            onPress={() => changeLanguage(language.code as 'es' | 'en')}
          >
            <Text style={styles.flag}>{language.flag}</Text>
            <Text
              style={[
                styles.languageName,
                locale === language.code && styles.selectedLanguageName,
              ]}
            >
              {language.name}
            </Text>
            {locale === language.code && (
              <Text style={styles.checkmark}>âœ“</Text>
            )}
          </TouchableOpacity>
        ))}
      </View>
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    padding: 16,
  },
  title: {
    fontSize: 18,
    fontWeight: 'bold',
    marginBottom: 16,
    color: '#333',
  },
  languageList: {
    gap: 8,
  },
  languageItem: {
    flexDirection: 'row',
    alignItems: 'center',
    padding: 12,
    borderRadius: 8,
    backgroundColor: '#f5f5f5',
    borderWidth: 1,
    borderColor: '#e0e0e0',
  },
  selectedLanguage: {
    backgroundColor: '#e3f2fd',
    borderColor: '#2196f3',
  },
  flag: {
    fontSize: 24,
    marginRight: 12,
  },
  languageName: {
    fontSize: 16,
    flex: 1,
    color: '#333',
  },
  selectedLanguageName: {
    color: '#2196f3',
    fontWeight: '600',
  },
  checkmark: {
    fontSize: 18,
    color: '#2196f3',
    fontWeight: 'bold',
  },
});
