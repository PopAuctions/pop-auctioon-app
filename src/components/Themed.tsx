/**
 * Learn more about Light and Dark modes:
 * https://docs.expo.io/guides/color-schemes/
 */

import { Text as DefaultText, View as DefaultView } from 'react-native';

import Colors from '@/constants/Colors';
import { useColorScheme } from './useColorScheme';

type ThemeProps = {
  lightColor?: string;
  darkColor?: string;
};

export type TextProps = ThemeProps & DefaultText['props'];
export type ViewProps = ThemeProps & DefaultView['props'];

export function useThemeColor(
  props: { light?: string; dark?: string },
  colorName: keyof typeof Colors.light & keyof typeof Colors.dark
) {
  const theme = useColorScheme() ?? 'light';
  const colorFromProps = props[theme];

  if (colorFromProps) {
    return colorFromProps;
  } else {
    return Colors[theme][colorName];
  }
}

export function Text(props: TextProps) {
  const { style, lightColor, darkColor, ...otherProps } = props;

  // Siempre llamar al hook, pero usar el resultado condicionalmente
  const themeColor = useThemeColor({ light: lightColor, dark: darkColor }, 'text');
  const shouldApplyThemeColor = lightColor || darkColor;
  const color = shouldApplyThemeColor ? themeColor : undefined;

  return <DefaultText style={[color ? { color } : {}, style]} {...otherProps} />;
}

export function View(props: ViewProps) {
  const { style, lightColor, darkColor, ...otherProps } = props;

  // Siempre llamar al hook, pero usar el resultado condicionalmente
  const themeBackgroundColor = useThemeColor({ light: lightColor, dark: darkColor }, 'background');
  const shouldApplyThemeColor = lightColor || darkColor;
  const backgroundColor = shouldApplyThemeColor ? themeBackgroundColor : undefined;

  return <DefaultView style={[backgroundColor ? { backgroundColor } : {}, style]} {...otherProps} />;
}