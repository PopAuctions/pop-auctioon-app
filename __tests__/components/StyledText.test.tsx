import * as React from 'react';
import { render } from '@testing-library/react-native';

import { MonoText } from '../../src/components/StyledText';

describe('MonoText Component', () => {
  it('renders correctly', () => {
    const { getByText } = render(<MonoText>Snapshot test!</MonoText>);

    // Verificar que el texto se renderiza correctamente
    expect(getByText('Snapshot test!')).toBeTruthy();
  });

  it('should match snapshot', () => {
    const { toJSON } = render(<MonoText>Snapshot test!</MonoText>);

    // Mantener el snapshot test usando la librería recomendada
    expect(toJSON()).toMatchSnapshot();
  });

  it('should render without crashing', () => {
    // Test básico de smoke test
    expect(() => render(<MonoText>Test content</MonoText>)).not.toThrow();
  });
});
