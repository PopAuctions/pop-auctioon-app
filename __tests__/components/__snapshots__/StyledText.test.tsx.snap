// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MonoText Component should match snapshot 1`] = `
<Text
  style={
    [
      {},
      [
        undefined,
        {
          "fontFamily": "SpaceMono",
        },
      ],
    ]
  }
>
  Snapshot test!
</Text>
`;
